ipset.go

```diff
@@ -154,6 +182,16 @@ var debugf = discardf
 // Ranges returns the minimum and sorted set of IP
 // ranges that covers s.
 func (s *IPSet) Ranges() []IPRange {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.rangesLocked()
+}
+
+func (s *IPSet) rangesLocked() []IPRange {
```

Comment:
I’d call this normalize, and then let callers use s.in after calling it. That makes it clearer that it mutates s.

Reply:
Using s.in is a bit weird, because unless you call normalize first, it's likely that you'll get the wrong answer. My goal was to encourage use of "what do you mean" accessors as much as possible. That, combined with the fact that Ranges has to do the computations required to normalize, made me combine in this way. WDYT?